uint32 sampleSize
---
bool success
string message